pkg/cmd/server/admin/create_nodeconfig.go

```diff
@@ -0,0 +1,376 @@
+			return err
+		}
+
+	} else {
+		if err := CopyFile(o.ClientCertFile, clientCertFile, 0644); err != nil {
+			return err
+		}
+		if err := CopyFile(o.ClientKeyFile, clientKeyFile, 0600); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o CreateNodeConfigOptions) MakeServerCert(serverCertFile, serverKeyFile string) error {
+	if o.IsCreateServerCertificate() {
+		nodeServerCertOptions := CreateServerCertOptions{
+			GetSignerCertOptions: o.GetSignerCertOptions,
+
+			CertFile: serverCertFile,
+			KeyFile:  serverKeyFile,
+
+			Hostnames: o.Hostnames,
+		}
+
+		if err := nodeServerCertOptions.Validate(nil); err != nil {
+			return err
+		}
+		if _, err := nodeServerCertOptions.CreateServerCert(); err != nil {
+			return err
+		}
+
+	} else {
+		if err := CopyFile(o.ServerCertFile, serverCertFile, 0644); err != nil {
+			return err
+		}
+		if err := CopyFile(o.ServerKeyFile, serverKeyFile, 0600); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o CreateNodeConfigOptions) MakeCA(clientCopyOfCAFile string) error {
+	if err := CopyFile(o.APIServerCAFile, clientCopyOfCAFile, 0644); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o CreateNodeConfigOptions) MakeKubeConfig(clientCertFile, clientKeyFile, clientCopyOfCAFile, kubeConfigFile string) error {
+	createKubeConfigOptions := CreateKubeConfigOptions{
+		APIServerURL:    o.APIServerURL,
+		APIServerCAFile: clientCopyOfCAFile,
+		ServerNick:      "master",
+
+		CertFile: clientCertFile,
+		KeyFile:  clientKeyFile,
+		UserNick: "node",
+
+		KubeConfigFile: kubeConfigFile,
+	}
+	if err := createKubeConfigOptions.Validate(nil); err != nil {
+		return err
+	}
+	if _, err := createKubeConfigOptions.CreateKubeConfig(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o CreateNodeConfigOptions) MakeNodeConfig(serverCertFile, serverKeyFile, kubeConfigFile, nodeConfigFile string) error {
+	config := &configapi.NodeConfig{
+		NodeName: o.NodeName,
+
+		ServingInfo: configapi.ServingInfo{
+			BindAddress: net.JoinHostPort(o.ListenAddr.Host, strconv.Itoa(ports.KubeletPort)),
+		},
+
+		VolumeDirectory:       o.VolumeDir,
+		NetworkContainerImage: o.NetworkContainerImage,
+		AllowDisabledDocker:   o.AllowDisabledDocker,
+
+		DNSDomain: o.DNSDomain,
+		DNSIP:     o.DNSIP,
+
+		MasterKubeConfig: kubeConfigFile,
+	}
+
+	if o.UseTLS() {
+		config.ServingInfo.ServerCert = configapi.CertInfo{
+			CertFile: serverCertFile,
+			KeyFile:  serverKeyFile,
+		}
+	}
+
+	// Resolve relative to CWD
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	if err := configapi.ResolveNodeConfigPaths(config, cwd); err != nil {
+		return err
+	}
+
+	// Relativize to config file dir
+	base, err := cmdutil.MakeAbs(o.NodeConfigDir, cwd)
+	if err != nil {
+		return err
+	}
+	if err := configapi.RelativizeNodeConfigPaths(config, base); err != nil {
+		return err
+	}
+	config.VolumeDirectory = o.VolumeDir // relativize is a little too aggressive
```

Comment:
You get the same thing depending on where you write the master config (e.g. `storageDirectory: ../openshift.local.etcd`).

Won't the `./foo` value be incorrect when we read in the node config and relativize all the paths? Seems like having *some* of the filepaths be relative to the config file is more confusing.